### PR TITLE
Collection runner upload data enhance[INS-4341]

### PR DESCRIPTION
**Pull request by cwangsmv**
Changes: 
- Different UI effect when user has uploaded data. 
Screenshot 2024-08-27 at 16 06 00Screenshot 2024-08-27 at 16 06 26
- Allow user to clear upload data 
Screenshot 2024-08-27 at 16 06 17

- Match last found request with same request name rather than first found request in collection runner
- Fix upload data preview table header upper case issue
- Add unit test for get iteration data value

### CODE REVIEW

**Review by filfreire — APPROVED** [review]

**Comment by jackkav** [comment]
Since this button doesn't match our apps general button conventions I thought I'd share my thoughts more specifically.


image


For this button and label. Since there are 3 possible actions.
- upload
- view
- discard

It might make sense to use a normal button with icon and label as we do in other places
image


Where initial state is upload icon and the label "Upload CSV"
Next state is eye icon and label "View CSV"
And possibly a trash can icon and "Discard CSV" button could appear beside View CSV


I don't think the green eye icon is necessary once the above changes are made. Let me know if I can help or be any clearer.

File: packages/insomnia/src/ui/components/modals/upload-runner-data-modal.tsx [hunk 1]
```diff
@@ -21,12 +21,13 @@ export interface UploadDataModalProps {
   onUploadFile: (file: File, data: UploadDataType[]) => void;
   onClose: () => void;
   userUploadData: UploadDataType[];
+  isPrview?: boolean;
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/modals/upload-runner-data-modal.tsx]: typo

**Comment by cwangsmv** [comment]
> Since this button doesn't match our apps general button conventions I thought I'd share my thoughts more specifically.
> 
> image
> For this button and label. Since there are 3 possible actions.
> 
> * upload
> * view
> * discard
> 
> It might make sense to use a normal button with icon and label as we do in other places image
> 
> Where initial state is upload icon and the label "Upload CSV" Next state is eye icon and label "View CSV" And possibly a trash can icon and "Discard CSV" button could appear beside View CSV
> 
> I don't think the green eye icon is necessary once the above changes are made. Let me know if I can help or be any clearer.

Yes, I think it's more clear. Due to the space is limited here, I will change to two states: **upload data** & **view data**.
When user views the data, he can also change to another file or clear the data.
Screenshot 2024-08-27 at 16 06 00Screenshot 2024-08-27 at 16 06 26
Screenshot 2024-08-27 at 16 06 17

File: packages/insomnia/src/ui/components/modals/upload-runner-data-modal.tsx (hunk 1 above)
Comment (cwangsmv) [reply packages/insomnia/src/ui/components/modals/upload-runner-data-modal.tsx]: Fixed

**Review by notjaywu — APPROVED** [review]
lgtm

**Pull request merged**